src/nm-ssh-service.c

```diff
@@ -515,7 +515,7 @@ send_network_config (NMSshPlugin *plugin)
 
 	/* replace default route? */
 	if (io_data->no_default_route) {
-		g_hash_table_insert (config, NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT, bool_to_gvalue (TRUE));
+		g_hash_table_insert (ip4config, NM_VPN_PLUGIN_IP4_CONFIG_NEVER_DEFAULT, bool_to_gvalue (TRUE));
```

Comment:
The change is working for me. Tested on Fedora 17 x86_64.
It looks like that ipv6 implementation has the same bug.
- g_hash_table_insert (config, NM_VPN_PLUGIN_CONFIG_HAS_IP6, bool_to_gvalue (TRUE));
+ g_hash_table_insert (ip6config, NM_VPN_PLUGIN_CONFIG_HAS_IP6, bool_to_gvalue (TRUE));
Do you want me to create one more pull request vor ipv6? I have no ipv6 setup net to try it.